File: DICCIONARIOS/Java/3.java
import java.util.*;

/**
 * Ejercicio 3 — Precios de repuestos de carro (Java)
 * Guarda un catálogo en un HashMap y calcula el total para un repuesto y cantidad dados.
 * Complejidad: búsquedas promedio O(1) por clave.
 */
public class Repuestos {
    public static void main(String[] args) {
        Map<String, Double> precios = new HashMap<>();
        precios.put("Bujia", 1.40);
        precios.put("Pastillas de freno", 5.80);
        precios.put("Radiador", 300.50);
        precios.put("Rodamiento", 10.70);

        Scanner sc = new Scanner(System.in);
        try {
            System.out.print("Ingrese el nombre del repuesto: ");
            String repuesto = sc.nextLine().trim();
            System.out.print("Ingrese la cantidad: ");
            int cantidad = Integer.parseInt(sc.nextLine().trim());

            Double precio = precios.get(repuesto);
            if (precio == null) {
                System.out.println("El repuesto no está disponible en el diccionario.");
            } else {
                double total = precio * cantidad;
                System.out.printf("El precio de %d %s(s) es: $%.2f%n", cantidad, repuesto, total);
            }
        } catch (NumberFormatException e) {
            System.out.println("Entrada inválida: la cantidad debe ser un entero.");
        }
    }
}

File: DICCIONARIOS/Python/3.py
#!/usr/bin/env python3
# Ejercicio 3 — Precios de repuestos de carro (Python)
# Programa que almacena un catálogo en un diccionario y calcula el total
# para un repuesto solicitado y una cantidad.
# Estructura: {nombre_repuesto: precio_unitario}
# Complejidad promedio de consulta: O(1) (acceso por clave).

PRECIOS = {
    "Bujia": 1.40,
    "Pastillas de freno": 5.80,
    "Radiador": 300.50,
    "Rodamiento": 10.70
}

def calcular_total(repuesto: str, cantidad: int) -> float | None:
    """Devuelve total precio si el repuesto existe; None en caso contrario."""
    precio = PRECIOS.get(repuesto)
    if precio is None:
        return None
    return precio * cantidad

if __name__ == "__main__":
    # Demostración interactiva; maneja errores de entrada.
    try:
        rep = input("Ingrese el nombre del repuesto: ").strip()
        cant = int(input("Ingrese la cantidad: ").strip())
        total = calcular_total(rep, cant)
        if total is None:
            print("El repuesto no está disponible en el diccionario.")
        else:
            print(f"El precio de {cant} {rep}(s) es: ${total:.2f}")
    except ValueError:
        print("Entrada inválida: la cantidad debe ser un número entero.")
